rest_api/src/main/java/com/scorelab/ioe/web/rest/PublicationResource.java

```diff
@@ -88,10 +105,35 @@
     public List<Publication> getAllPublications() {
         log.debug("REST request to get all Publications");
         List<Publication> publications = publicationRepository.findAll();
+
         return publications;
     }
 
     /**
+     * POST /publish/:sensorid
+     * Insert sensor data to Cassandra storage
+     *
+     * @throws URISyntaxException if the Location URI syntax is incorrect
+     */
+
+    @RequestMapping(value = "/publish/{sensorid}",
+    	method = RequestMethod.POST,
+        produces = MediaType.APPLICATION_JSON_VALUE)
+
+    @Timed
+    public void publishData (@Valid @RequestBody SensorData sensorData){
+        List<Publication> publications = publicationRepository.findAll();
```

Comment:
Can't you query directly using the sensor ID instead of retrieving all publications from DB and then filtering it in the server?